modules/tinymce/src/plugins/table/main/ts/actions/TableActions.ts

```diff
@@ -101,6 +112,50 @@ export const TableActions = (editor: Editor, lazyWire: () => ResizeWire): TableA
 
   const pasteCells = execute(TableOperations.pasteCells, Fun.always, Fun.noop, lazyWire);
 
+  const setTableCellType = (editor: Editor, args: Record<string, any>) =>
+    Obj.get(args, 'type').each((type) => {
+      if (Arr.contains([ 'td', 'th' ], type)) {
+        switchCellType(editor.dom, getCellsFromSelection(editor), type);
+      }
+    });
+
+  const setTableRowType = (editor: Editor, args: Record<string, any>) =>
+    // type: 'header' | 'body' | 'footer'
+    Obj.get(args, 'type').each((type) => {
```

Comment:
This might be a little nicer/cleaner if you use `.filter` to filter out the invalid type maybe? eg:

```
const extractType = (args: Record<string, any>, validTypes: string[]) =>
  Obj.get(args, 'type').filter((type) => Arr.contains(validTypes, type));

const setTableCellType = (editor: Editor, args: Record<string, any>) =>
  extractType(args, [ 'td', 'th' ]).each((type) => {
    switchCellType(editor.dom, getCellsFromSelection(editor), type);
  });

const setTableRowType = (editor: Editor, args: Record<string, any>) =>
  extractType(args, [ 'header', 'body', 'footer' ]).each((type) => {
    Arr.map(getRowsFromSelection(editor), (row) => switchSectionType(editor, row, type));
  });
```
and that could be used here and for the cell type.